crates/core/src/error/collector.rs

```diff
@@ -0,0 +1,289 @@
+use derive_more::From;
+
+use crate::{error::RusticErrorKind, RusticError};
+use log::{error, info, warn};
+
+/// A rustic issue result
+///
+/// rustic issue results are used to return a result along with possible issues.
+#[derive(Debug)]
+pub enum RusticIssueResult<T> {
+    /// For when we return a result along with possible issues
+    Ok(T, Option<Vec<RusticIssue>>),
+
+    /// For when we abort and return only the errors that occurred
+    Err(Vec<RusticError>),
+}
```

Comment:
This is actually equivalent to `Result<(T, Option<Vec<RusticIssue>>), Vec<RusticError>>`

Besides this, I would not us `Option<Vec<..>>` but only `Vec<..>` and treat an empty `Vec` like the `None` case. I think this simplifies implementation

Reply:
Yes, still a draft and I was sketching out that type and landed there 😅